week2/homework/src/removeList.js

```diff
@@ -0,0 +1,19 @@
+const fs = require('fs');
+const colors = require('colors');
+function removeList(index) {
+  try {
+    if (index <= 0 || index > toDosArray.length) {
+      console.log(`You've got ${toDosArray.length} list/s`);
```

Comment:
this will show the message but it won't stop, but it will try to remove something isn't exist or it's not valid, so it may give an error.

Reply:
I didn't get this point! Can you clarify it again? Maybe you meant to get if statement out of try?

Reply:
I mean you have an invalid index, and you are making that clear to the user by telling him how many items he have.

and then you are reading the file on line 8 and taking of the item on line 9 even it is not valid item and it will be an `undefined` value, and you completing the process till writing `undefined` to the array.

Reply:
I think I fixed the problem! 
try {
    const toDosArray = JSON.parse(fs.readFileSync('toDoList.json'));
    const item = toDosArray[index - 1];
    const lists = toDosArray.filter(list => list !== item);
    fs.writeFileSync('./toDoList.json', JSON.stringify(lists));
    if (index <= 0 || index > toDosArray.length) {
      console.log(`You've got ${toDosArray.length} list/s`);
    }
  } catch (error) {
    console.log(colors.yellow(`${error}`));
  }

Reply:
that will solve the error you have but it will stay writing unnecessary `undefined` to the file you have and it may cause an error since it should be writing something to the file.

avoiding any unnecessary stuff you may do the following:
```javascript
try {
  const toDosArray = JSON.parse(fs.readFileSync('toDoList.json'));
  if (index <= 0 || index > toDosArray.length) {
    console.log(You've got ${toDosArray.length} list/s);
  } else {
    // this will prevent any redundant actions
    const item = toDosArray[index - 1];
    const lists = toDosArray.filter(list => list !== item);
    fs.writeFileSync('./toDoList.json', JSON.stringify(lists));
  }
} catch (error) {
  console.log(colors.yellow(${error}));
}
```